### DISCUSSION TITLE
Needs help understanding cache & using KV as cache

### DISCUSSION THREAD

**Discussion by ptgamr** [cloudflare/next-on-pages#750] [General]
Hi there, 

I have a brand new `next-on-pages` project and managed to deploy it. However, I'm struggling to understand how the Cache works.

Let say I have an `/api/something/route.ts`, which initiate a `fetch('some-external-url')`

Am I correct understanding that fetch will be cached using the CacheAPI? ... and if so, for how long? What is the default value for TTL?

Also, I tried to use `KV` binding and have these in my `wranger.toml` file:

```
[[kv_namespaces]]
binding = "__NEXT_ON_PAGES__KV_SUSPENSE_CACHE"
id = "*********"
```

I would expect I can then inspect what is cached inside that KV namespace... but no matter how many requests I've made, I see this namespace still empty.

So at this point, I'm not totally sure if the cache is working at all...

Thanks for your help,
Anh

**Comment by exectx** [comment]
It works for me (prod and local).
For ttl you can check this for cache api: https://github.com/cloudflare/next-on-pages/blob/317e872bbfbc4a2ee844618364cab1bc6141ed56/packages/next-on-pages/templates/cache/cache-api.ts#L23

**Comment by r1tsuu** [comment]
It doesn't work for me too.. no matter what versions of next / @cloudflrare/next-on-pages i installed.. KV is just empty always and by speed i can determine as well that caching doesn't work
